include/xtensor-zarr/xzarr_hierarchy.hpp

```diff
@@ -0,0 +1,157 @@
+
+        template <class value_type, class io_handler = xdisk_io_handler<xblosc_config>>
+        auto get_array(const char* path);
+
+    private:
+        fs::path m_path;
+    };
+
+    void xzarr_hierarchy::create_hierarchy()
+    {
+        nlohmann::json j;
+        j["zarr_format"] = "https://purl.org/zarr/spec/protocol/core/3.0";
+        j["metadata_encoding"] = "application/json";
+        j["extensions"] = nlohmann::json::array();
+
+        fs::create_directories(m_path);
+        auto zarr_json = m_path / "zarr.json";
+        std::ofstream o(zarr_json.string());
+        o << std::setw(4) << j << std::endl;
+    }
+
+    template <class value_type, class shape_type, class io_handler = xdisk_io_handler<xblosc_config>>
+    auto xzarr_hierarchy::create_array(const char* path, shape_type shape, shape_type chunk_shape, nlohmann::json& attrs)
+    {
+        auto meta_path = get_meta_path(m_path, path);
+        auto meta_path_array = meta_path;
+        meta_path_array += ".array";
+        auto data_path = get_data_path(m_path, path);
+
+        nlohmann::json j;
+        j["shape"] = shape;
+        j["chunk_grid"]["type"] = "regular";
+        j["chunk_grid"]["chunk_shape"] = chunk_shape;
+        j["attributes"] = attrs;
+        std::ofstream stream(meta_path_array);
+        stream << std::setw(4) << j << std::endl;
+
+        xchunked_array<xchunk_store_manager<xfile_array<value_type, io_handler>>, xzarr_attrs> a(shape, chunk_shape);
+        a.chunks().set_directory(data_path.string().c_str());
+        return a;
+    }
+
+    template <class value_type, class io_handler = xdisk_io_handler<xblosc_config>>
+    auto xzarr_hierarchy::get_array(const char* path)
+    {
+        int i;
+        auto meta_path = get_meta_path(m_path, path);
+        auto meta_path_array = meta_path;
+        meta_path_array += ".array";
+        auto data_path = get_data_path(m_path, path);
+
+        std::ifstream stream(meta_path_array);
+        std::string s;
+        stream.seekg(0, stream.end);
+        s.reserve(stream.tellg());
+        stream.seekg(0, stream.beg);
+        s.assign((std::istreambuf_iterator<char>(stream)),
+                  std::istreambuf_iterator<char>());
+        auto j = nlohmann::json::parse(s);
+        auto json_shape = j["shape"];
+        auto json_chunk_shape = j["chunk_grid"]["chunk_shape"];
+        std::vector<std::size_t> shape(json_shape.size());
+        std::vector<std::size_t> chunk_shape(json_chunk_shape.size());
+        i = 0;
+        for (auto size: json_shape)
+        {
+            shape[i] = stoi(size.dump(), nullptr);
+            i++;
+        }
+        i = 0;
+        for (auto size: json_chunk_shape)
+        {
+            chunk_shape[i] = stoi(size.dump(), nullptr);
+            i++;
+        }
+        xchunked_array<xchunk_store_manager<xfile_array<value_type, io_handler>>, xzarr_attrs> a(shape, chunk_shape);
+        a.chunks().set_directory(data_path.string().c_str());
```

Comment:
I think the directory should be an argument of `xchunked_array` constructor (I'll open an issue in xtensor to discuss this).

Reply:
I'm not sure, a directory really makes sense only for `xchunk_store_manager`, and not for an in-memory `xchunked_array`.

Reply:
Ah indeed, I thought it was already accessible via the `xchunker_array`. Let's keep it as is then.